### ISSUE TITLE
Face detection : Unhandled Exception: type 'Null' is not a subtype of type 'Iterable<dynamic>'

### ISSUE THREAD

**Issue by nicolasvanheusden** [AgoraIO-Extensions/Agora-Flutter-SDK#752]
**Describe the bug**
The Exception : Unhandled Exception: type 'Null' is not a subtype of type 'Iterable<dynamic>' is throw when listening to facePositionChanged  from RtcEngineEventHandler occurs multiple times.

**Minimal reproduction code**
```dart
import 'package:agora_rtc_engine/rtc_engine.dart';
import 'package:agora_test/app_id.dart';
import 'package:flutter/material.dart';
import 'package:agora_rtc_engine/rtc_local_view.dart' as RtcLocalView;

class HomePage extends StatefulWidget {
  const HomePage({Key? key}) : super(key: key);

  @override
  State<HomePage> createState() => _HomePageState();
}

class _HomePageState extends State<HomePage> {

  late RtcEngine _engine;
  late int? streamId;

  @override
  void initState() {    
    initAgora();
    super.initState();
  }

  Future<void> initAgora() async {

    await _initAgoraRtcEngine();
    await _engine.createDataStream(false, false);

    _engine.setEventHandler(RtcEngineEventHandler(
      joinChannelSuccess: (channel, uid, elapsed) {
        setState(() {
          print('onJoinChannel: $channel, uid: $uid');
        });
      },
      leaveChannel: (stats) {
        setState(() {
          print('onLeaveChannel');
        });
      },
      userJoined: (uid, elapsed) {
        setState(() {
          print('userJoined: $uid');
        });
      },
      userOffline: (uid, elapsed) {
        setState(() {
          print('userOffline: $uid');
        });
      },
      streamMessage: (_, __, message) {
        final String info = "here is the message $message";
        print(info);
      },
      streamMessageError: (_, __, error, ___, ____) {
        final String info = "here is the error $error";
        print(info);
      },
      // added
      facePositionChanged: (imageWidth, imageHeight, faces) {
        if (faces.isEmpty) {
          print("no faces");
        } else {
          print("here you are");
        }        
      }
    ));   
    await _engine.joinChannel(token, channel, null, 0);
    
  }


  Future<void> _initAgoraRtcEngine() async {
    _engine = await RtcEngine.create(appid);
    await _engine.enableVideo();
    await _engine.setChannelProfile(ChannelProfile.LiveBroadcasting);
    await _engine.setClientRole(ClientRole.Broadcaster);
    await _engine.startPreview(); // added
    await _engine.enableFaceDetection(true); // added
  }



  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Stack(
        children: [
          const RtcLocalView.SurfaceView(),
          Align(
            alignment: Alignment.bottomCenter,
            child: ElevatedButton.icon(
              onPressed: () {
                _engine.leaveChannel();
              },
              icon: const Icon(Icons.call_end_rounded),
              label: const Text('close')
            ),
          ),
        ],
      )
    );
  }
}
```

**Version :**
- agora_rtc_engine: ^5.3.0
- flutter 3.3.0

On IOS, It seems to fail when no faces are detected. 
But on Android, it seems to fail randomly OR it's just the exception  that is thrown after that a face was detected due to multithreading maybe ?

**Comment by littleGnAl** [comment]
Thanks for reporting, it will be fixed on the next release.

**Comment by littleGnAl** [comment]
This fix has been merged to https://github.com/AgoraIO-Extensions/Agora-Flutter-SDK/tree/master, you can depend on the `agora_rtc_engine` as the git package at this time.

**Comment by littleGnAl** [comment]
This fix has been released `agora_rtc_engine: ^5.3.1`.
https://pub.dev/packages/agora_rtc_engine
